src/torrent.cpp

```diff
@@ -1479,37 +1538,20 @@ bool is_downloading_state(int const st)
 
 	void torrent::init_ssl(string_view cert)
 	{
-		using boost::asio::ssl::context;
-
-		// this is needed for openssl < 1.0 to decrypt keys created by openssl 1.0+
-#if !defined(OPENSSL_API_COMPAT) || (OPENSSL_API_COMPAT < 0x10100000L)
-		OpenSSL_add_all_algorithms();
-#else
-		OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_CIPHERS | OPENSSL_INIT_ADD_ALL_DIGESTS, nullptr);
-#endif
-
 		// create the SSL context for this torrent. We need to
 		// inject the root certificate, and no other, to
 		// verify other peers against
-		std::unique_ptr<context> ctx(std::make_unique<context>(context::tls));
-
-		if (!ctx)
-		{
-			error_code ec(int(::ERR_get_error()),
-				boost::asio::error::get_ssl_category());
-			set_error(ec, torrent_status::error_file_ssl_ctx);
-			pause();
-			return;
-		}
+		std::unique_ptr<ssl::context> ctx(std::make_unique<ssl::context>(ssl::context::tls));
 
-		ctx->set_options(context::default_workarounds
-			| context::no_sslv2
-			| context::single_dh_use);
+		ctx->set_options(ssl::context::default_workarounds
+			| ssl::context::no_sslv2
+			| ssl::context::no_sslv3
```

Comment:
SSLv3 is outdated an insecure by now I assume?

Reply:
Indeed, anyway this is somewhat redundant since now the method is set to `tls`.